### ISSUE TITLE
UTF-8 "\xF3" does not map to Unicode at /usr/share/perl5/vendor_perl/Locale/Po4a/TransTractor.pm line 583

### ISSUE THREAD

**Issue by sergiomb2** [mquinson/po4a#477]
debhelper-13.11.6-3.fc41 FTBFS: UTF-8 "\xF3" does not map to Unicode at /usr/share/perl5/vendor_perl/Locale/Po4a/TransTractor.pm line 583

This failure is probably triggered by upgrading po4a from 0.69-5.fc40 to 0.70-2.fc41 

https://bugzilla.redhat.com/show_bug.cgi?id=2266008 

you may check the builds on https://koschei.fedoraproject.org/package/debhelper , particularly  https://koschei.fedoraproject.org/build/17470881

**Comment by Fat-Zer** [comment]
> I don't see this char in the source code. Help is welcome, please.

it's in [spanish addendums][1], which are in ISO-8856-15 charset. It's `ó` in `Traducción`.

It seems they correctly specify the addendum charset in [po4a][2] for Spanish, but po4a tries to read it in UTF-8 anyway... so it's probably a po4a's bug.

Besides that po4a probably should print some less cryptic error message with a reference to the file at least and maybe continue with the rest of the documents instead of straight up crashing...

To reproduce:

```
git clone https://salsa.debian.org/debian/debhelper.git
cd debhelper
po4a man/po4a/po4a.cfg
```


[1]: https://salsa.debian.org/debian/debhelper/-/blob/main/man/po4a/add1.es?ref_type=heads
[2]: https://salsa.debian.org/debian/debhelper/-/blob/main/man/po4a/po4a.cfg?ref_type=heads#L4

**Comment by mquinson** [comment]
Ok, I think I got it. I must have forgotten to use PerlIO on the filehandle that is in charge of the addendum. I'll look at it later on.

Many many thanks @Fat-Zer for this great diagnostic. You really rock.

**Comment by mquinson** [comment]
Ok, I nailed this down, I guess. The main issue was the po4a.cfg file which is not nice with us. 

The first problem is that it's redefining the pod alias several times. I think that this was already broken in the past (only the last definition is kept) but before the introduction of PerlIO, the `-A ISO-8859-15` was useless for some reason involving Perl trying to magically guess what to do.

I fixed this in commit 300208e197ce4e8a7692ed835f093abb8c07007b, but debhelper still fails with this error:
```
Malformed encoding while writing char '–' to file /tmp/debhelper/man/de/debhelper.pod with charset ISO-8859-15: "\x{2013}" does not map to iso-8859-15 at 
```

This is because of the `-L iso-8859-15` in the config file, which shouldn't be there since most languages of the list use UTF-8 chars that cannot be represented in Latin1. It used to work because all those -L parameters where wrongly discarded because of the previous bug.

I think that my code fixes this bug, but I leave it open for further discussion in case someone has a better idea.
